digital_asset_types/src/dao/scopes/asset.rs

```diff
@@ -409,3 +413,63 @@ async fn get_grand_total(
     let grand_total = stmt.count(conn).await?;
     Ok(Some(grand_total))
 }
+
+pub async fn add_collection_metadata(
+    conn: &impl ConnectionTrait,
+    mut asset_list: AssetList,
+) -> Result<AssetList, DbErr> {
+    // compile a vec of all the group values (String) from the asset list
+    let mut group_values: Vec<String> = Vec::new();
+    for item in &asset_list.items {
+        if let Some(groups) = item.grouping.clone() {
+            for group in groups {
+                let group_value = group.group_value;
+                if let Some(group_value) = group_value {
+                    group_values.push(group_value);
+                }
+            }
+        }
+    }
+
+    // convert the group values to bytea (bc asset_data.id is bytea)
+    let bytea_group_values: Vec<Vec<u8>> = group_values
+        .iter()
+        .map(|group_value| {
+            let bytea_group_value = group_value.as_bytes().to_vec();
```

Comment:
that's not gonna work i believe. the string is bs58 encoded. You need to bs58 decode the string and then pass that in